src/tokens/blockchain.service.ts

```diff
@@ -88,15 +104,60 @@ export class BlockchainConnectorService {
     });
   }
 
+  // Check if retry condition matches the err that's been hit
+  private matchesRetryCondition(err: any): boolean {
+    return this.retryCondition != '' && err?.toString().match(this.retryCondition) !== null;
+  }
+
+  // Delay by the appropriate amount of time given the iteration the caller is in
+  private async backoffDelay(iteration: number) {
+    const delay = Math.min(
+      this.retryBackOffInitial * Math.pow(this.retryBackOffFactor, iteration),
+      this.retryBackOffLimit,
+    );
+    await new Promise(resolve => setTimeout(resolve, delay));
+  }
+
+  // Generic helper function that makes an given blockchain function retryable
+  // by using synchronous, back-off delays for cases where the function returns
+  // an error which matches the configured retry condition
+  private async retryableCall<T = any>(
+    blockchainFunction: () => Promise<AxiosResponse<T>>,
+  ): Promise<AxiosResponse<T>> {
+    let response: any;
+    for (let retries = 0; retries <= this.retriesMax; retries++) {
```

Comment:
Will this end up allowing `retriesMax+1` attempts? ie if I set it to 15, it will actually attempt the call 16 times?

Reply:
Good point, I'll add it to the doc